code/modules/detectivework/scanner.dm

```diff
@@ -14,12 +14,48 @@
 	flags_1 = CONDUCT_1
 	item_flags = NOBLUDGEON
 	slot_flags = ITEM_SLOT_BELT
+	actions_types = list(/datum/action/item_action/displayDetectiveScanResults)
+	var/icons_available = list() // stores available icons for radial menu
+	var/icon_directory = 'icons/effects/icons.dmi' // dmi file containing the icons used in the radial menu
 	var/scanning = 0
+	var/found_something // placeholder for result boolian. placed here for admin scanner.
 	var/list/log = list()
 	var/range = 8
 	var/view_check = TRUE
 	var/forensicPrintCount = 0
-	actions_types = list(/datum/action/item_action/displayDetectiveScanResults)
+	var/icon_state_neutral // stores the neutral icon state
+	var/scan_icon = TRUE // does the forensic scanner have a scanning icon state?
+	var/icon_state_scanning =  "forensicnew_scan" // icon state for scanning
+	var/scan_speed = 4 // the speed between scans. The delay is this value x2 . Total scan time  = scan_speed * 10 for minimum , scan_speed * 18 if results for each catagory are found
+	var/admin = FALSE // does this scanner pull up hiddenprints? Also removes scan time and feedback giving instant results
+	var/advanced = FALSE // does this scanner pull up more details on results?
+
+	// sounds must be in sound/items by default. Modify sound_directory for a different directory.
+	var/can_sound = TRUE // can this scanner play sound at all?
+	var/sound_on = TRUE // is the sound currently turned on?
+	var/sound_directory = "sound/items" // just in-case you decide to place your sounds outside of sound\items
+	var/sound_positive = "scanner_positive" // plays as a result is found during the scan
+	var/sound_scan = "scanner_scan" // plays as each new scan is starting
+	var/sound_match = "scanner_match" // plays at the end when there is a match found
+	var/sound_nomatch = "scanner_nomatch" // plays at the end when no matches were found
```

Comment:
Remove the dynamic sound filename thing, this will cause lag whenever someone uses the scanner for the first time as the resources wont be preloaded. Also this is a minor security risk thats really not needed.

Reply:
Modified. 
r.i.p being able to edit the sounds in VV during the shift